src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,207 @@
+            _autoLogonRegManager = HostContext.GetService<IAutoLogonRegistryManager>();
+            _store = hostContext.GetService<IConfigurationStore>();
+        }
+
+        public async Task Configure(CommandSettings command)
+        {
+            if (!_windowsServiceHelper.IsRunningInElevatedMode())
+            {
+                Trace.Error("Needs Administrator privileges to configure agent with AutoLogon capability.");
+                Trace.Error("You will need to unconfigure the agent and then re-configure with Administrative rights");            
+                throw new SecurityException(StringUtil.Loc("NeedAdminForAutologonCapability"));
+            }
+
+            string domainName;
+            string userName;
+            string logonAccount;
+            string logonPassword;
+
+            while (true)
+            {
+                logonAccount = command.GetWindowsLogonAccount(defaultValue: string.Empty, descriptionMsg: StringUtil.Loc("AutoLogonAccountNameDescription"));
+                GetAccountSegments(logonAccount, out domainName, out userName);
+
+                if ((string.IsNullOrEmpty(domainName) || domainName.Equals(".", StringComparison.CurrentCultureIgnoreCase)) && !logonAccount.Contains("@"))
+                {
+                    logonAccount = String.Format("{0}\\{1}", Environment.MachineName, userName);
+                }
+                Trace.Info("LogonAccount after transforming: {0}, user: {1}, domain: {2}", logonAccount, userName, domainName);
+
+                logonPassword = command.GetWindowsLogonPassword(logonAccount);
+                if (_windowsServiceHelper.IsValidCredential(domainName, userName, logonPassword))
+                {
+                    Trace.Info("Credential validation succeeded");
+                    break;
+                }
+                
+                if (command.Unattended)
+                {
+                    throw new SecurityException(StringUtil.Loc("InvalidWindowsCredential"));
+                }
+                    
+                Trace.Error("Invalid credential entered");
+                _terminal.WriteLine(StringUtil.Loc("InvalidWindowsCredential"));
+            }
+
+            bool isCurrentUserSameAsAutoLogonUser = _windowsServiceHelper.HasActiveSession(domainName, userName);            
+            if (!isCurrentUserSameAsAutoLogonUser)
+            {
+                if (!_autoLogonRegManager.DoesRegistryExistForUser(domainName, userName))
+                {
+                    Trace.Error($"The autologon user '{logonAccount}' doesnt have a user profile on the machine. Please login once with the expected autologon user and reconfigure the agent again");
+                    throw new InvalidOperationException(StringUtil.Loc("NoUserProfile", logonAccount));
+                }
+            }
+            _autoLogonRegManager.LogWarnings(domainName, userName);
```

Comment:
you are not log warnings, you are inspect current registry setting for the given user.